server/util/log/log.go

```diff
@@ -64,13 +66,106 @@ func LogGRPCRequest(ctx context.Context, fullMethod string, dur time.Duration, e
 	reqID, _ := uuid.GetFromContext(ctx) // Ignore error, we're logging anyway.
-	log.Printf("HTTP %s %q %d %s [%s]", reqID, url, statusCode, http.StatusText(statusCode), formatDuration(dur))
+	Printf("HTTP %s %q %d %s [%s]", reqID, url, statusCode, http.StatusText(statusCode), formatDuration(dur))
+}
+
+func Configure(level string, enableStructured bool) error {
+	fmt.Printf("Configured called: level: %q, enableStructured: %t\n", level, enableStructured)
+	if enableStructured {
+		log.Logger = StructuredLogger()
+	} else {
+		log.Logger = LocalLogger(level)
+	}
+	intLogLevel := zerolog.InfoLevel
+	if level != "" {
+		if l, err := zerolog.ParseLevel(level); err == nil {
+			intLogLevel = l
+		} else {
+			return err
+		}
+	}
+	zerolog.SetGlobalLevel(intLogLevel)
+	return nil
+}
+
+func LocalLogger(level string) zerolog.Logger {
+	output := zerolog.ConsoleWriter{Out: os.Stdout}
+	// Skipping 3 frames prints the correct source file + line number, rather
+	// than printing a line number in this file or in the zerolog library.
+	return zerolog.New(output).With().Timestamp().Logger().With().CallerWithSkipFrameCount(3).Logger()
+}
+
+func StructuredLogger() zerolog.Logger {
+	// These overrides configure the logger to emit structured
+	// events compatible with GCP's logging infrastructure.
+	zerolog.LevelFieldName = "severity"
+	zerolog.TimestampFieldName = "timestamp"
+	zerolog.TimeFieldFormat = time.RFC3339Nano
+	return log.Logger
+}
+
+// Zerolog convenience wrapper below here:
+
+// DEPRECATED: use log.Info instead!
+func Print(message string) {
+	log.Info().Msg(message)
+}
+
+// DEPRECATED: use log.Infof instead!
+func Printf(format string, v ...interface{}) {
+	log.Info().Msgf(format, v...)
+}
+
+// Info logs to the INFO log.
+func Info(message string) {
+	log.Info().Msg(message)
+}
+
+// Infof logs to the INFO log. Arguments are handled in the manner of fmt.Printf.
+func Infof(format string, args ...interface{}) {
+	log.Info().Msgf(format, args...)
+}
+
+// Warning logs to the WARNING log.
+func Warning(message string) {
+	log.Warn().Msg(message)
+}
+
+// Warningf logs to the WARNING log. Arguments are handled in the manner of fmt.Printf.
+func Warningf(format string, args ...interface{}) {
+	log.Warn().Msgf(format, args...)
+}
+
+// Error logs to the ERROR log.
+func Error(message string) {
+	log.Error().Msg(message)
+}
+
+// Errorf logs to the ERROR log. Arguments are handled in the manner of fmt.Printf.
+func Errorf(format string, args ...interface{}) {
+	log.Error().Msgf(format, args...)
+}
+
+// Fatal logs to the FATAL log. Arguments are handled in the manner of fmt.Print.
+// It calls os.Exit() with exit code 1.
+func Fatal(message string) {
+	log.Fatal().Msg(message)
+	// Make sure fatal logs will exit.
+	os.Exit(1)
+}
+
+// Fatalf logs to the FATAL log. Arguments are handled in the manner of fmt.Printf.
+// It calls os.Exit() with exit code 1.
+func Fatalf(format string, args ...interface{}) {
```

Comment:
What do you think about allowing Debug level logs for logs that are useful for debugging tests but are too noisy to be used in production?

Reply:
Yeah good catch, I missed debug :) Added!